runtime/src/main/jni/NetworkDomainCallbackHandlers.h

```diff
@@ -0,0 +1,278 @@
+        auto hasTextContentBool = hasTextContent->BooleanValue();
+
+        auto responses = networkAgentInstance->m_responses;
+        auto it = responses.find(requestIdString);
+
+        if (it == responses.end()) {
+            throw NativeScriptException("Response not found for requestId = " +
+                                        ArgConverter::ConvertToString(requestId));
+        } else {
+            v8_inspector::utils::NetworkRequestData* response = it->second;
+
+            if (!hasTextContentBool) {
+                response->setData(dataString);
+            } else {
+                response->setData(ArgConverter::ConvertToString(data));
+            }
+
+            response->setHasTextContent(hasTextContentBool);
+        }
+    } catch (NativeScriptException& e) {
+        e.ReThrowToV8();
+    } catch (std::exception e) {
+        std::stringstream ss;
+        ss << "Error: c exception: " << e.what() << std::endl;
+        NativeScriptException nsEx(ss.str());
+        nsEx.ReThrowToV8();
+    } catch (...) {
+        NativeScriptException nsEx(std::string("Error: c exception!"));
+        nsEx.ReThrowToV8();
+    }
+}
+
+static void LoadingFinishedCallback(const v8::FunctionCallbackInfo<v8::Value>& args) {
+    try {
+        auto networkAgentInstance = V8NetworkAgentImpl::Instance;
+
+        if (!networkAgentInstance) {
+            return;
+        }
+
+        if (args.Length() != 0 && !args[0]->IsObject()) {
+            return;
+        }
+
+        auto isolate = args.GetIsolate();
+
+        v8::HandleScope scope(isolate);
+
+        auto context = isolate->GetCurrentContext();
+
+        v8::Local<v8::Object> argsObj = args[0]->ToObject();
+
+        if (!argsObj->Has(context, ArgConverter::ConvertToV8String(isolate, "requestId")).FromMaybe(false) ||
+                !argsObj->Has(context, ArgConverter::ConvertToV8String(isolate, "timestamp")).FromMaybe(false)) {
+            throw NativeScriptException(
+                "Not all parameters are present in the object argument in the call to LoadingFinished! Required params: 'requestId', `timeStamp`");
+        }
+
+        auto requestId = argsObj->Get(context, ArgConverter::ConvertToV8String(isolate, "requestId")).ToLocalChecked()->ToString();
+        auto timeStamp = argsObj->Get(context, ArgConverter::ConvertToV8String(isolate, "timestamp")).ToLocalChecked()->ToNumber()->IntegerValue();
+
+        networkAgentInstance->m_frontend.loadingFinished(ArgConverter::ConvertToString(requestId).c_str(), timeStamp);
+    } catch (NativeScriptException& e) {
+        e.ReThrowToV8();
+    } catch (std::exception e) {
+        std::stringstream ss;
+        ss << "Error: c exception: " << e.what() << std::endl;
+        NativeScriptException nsEx(ss.str());
+        nsEx.ReThrowToV8();
+    } catch (...) {
+        NativeScriptException nsEx(std::string("Error: c exception!"));
+        nsEx.ReThrowToV8();
+    }
+}
+}
```

Comment:
Code is formatted according to the tool used - astyle. https://github.com/NativeScript/android-runtime/blob/master/tools/astyle.js#L11